### PR TITLE
Modifies DATE/TIME tests to assert by Ion and modifies TIME output

### LINKED ISSUE #567: CAST tests for DATE values assert expected results using string equality instead of Ion equality
Note the call to `.toString`, at the end of the line here:

https://github.com/partiql/partiql-lang-kotlin/blob/fe306c6cf47c61e0fd31693241c2c987651a7753/lang/test/org/partiql/lang/eval/CastTestBase.kt#L217

This is a sub-optimal method of performing assertions on expected results--actually, the result should be specified in Ion, and the assertion should be performed on the resulting Ion values.

Something like:

```
assertEquals(ion.singleValue(castCase.expected), eval(castCase.expression, compileOptions = CompileOptions.build(compileOptionBlock)).ionValue)
```

For this to work, there's a couple hundred test cases here that need their expected values converted to Ion.

### PR DESCRIPTION
**Pull request by johnedquinn**
## Relevant Issue
- Closes #567 

## Description
- Modifies the tests to not compare by string but by ExprValue
- Modifies the output string of TIME
  - In order to compare by ExprValue, we need to make sure that we retain offset via TIME WITH TIME ZONE

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lang/test/org/partiql/lang/eval/EvaluatingCompilerCastTest.kt [hunk 1]
```diff
@@ -35,6 +35,6 @@ class EvaluatingCompilerCastTest : CastTestBase() {
     @ArgumentsSource(DateTimeConfiguredCastArguments::class)
     fun dateTimeConfiguredCast(dateTimeConfiguredCastCase: ConfiguredCastCase) =
         dateTimeConfiguredCastCase.assertCase(
-            ExpectedResultFormat.STRING
+            ExpectedResultFormat.PARTIQL
```
Comment (dlurton) [inline lang/test/org/partiql/lang/eval/EvaluatingCompilerCastTest.kt:38]: While this is an improvement over string equality and is perfectly acceptable for `TIME` types, I might have considered `ExpectedResultFormat.ION` instead. See https://github.com/partiql/partiql-lang-kotlin/issues/560 for an explanation as to why none of the expected value formats in use today are truly ideal.  However, `ExpectedResultFormat.ION` is usually the most ideal one available today.

File: lang/test/org/partiql/lang/eval/CastTestBase.kt [hunk 2]
```diff
@@ -1359,6 +1344,21 @@ abstract class CastTestBase : EvaluatorTestBase() {
...
+                case("TIME '23:12:12.1267+05:30'", "\"23:12:12.1267\"", CastQuality.LOSSLESS),
+                case("TIME (3) '23:12:12.1267'", "\"23:12:12.127\"", CastQuality.LOSSLESS),
+                case("TIME (3) '23:12:12.1267-05:30'", "\"23:12:12.127\"", CastQuality.LOSSLESS),
+                case("TIME (3) '23:12:12.1267+05:30'", "\"23:12:12.127\"", CastQuality.LOSSLESS),
+                case("TIME WITH TIME ZONE '23:12:12.1267'", "\"23:12:12.1267${defaultTimezoneOffset.getOffsetHHmm()}\"", CastQuality.LOSSLESS),
+                case("TIME WITH TIME ZONE '23:12:12.1267-05:30'", "\"23:12:12.1267-05:30\"", CastQuality.LOSSLESS),
+                case("TIME WITH TIME ZONE '23:12:12.1267+05:30'", "\"23:12:12.1267+05:30\"", CastQuality.LOSSLESS),
+                case("TIME (3) WITH TIME ZONE '23:12:12.1267'", "\"23:12:12.127${defaultTimezoneOffset.getOffsetHHmm()}\"", CastQuality.LOSSLESS),
+                case("TIME (3) WITH TIME ZONE '23:12:12.1267-05:30'", "\"23:12:12.127-05:30\"", CastQuality.LOSSLESS),
+                case("TIME (3) WITH TIME ZONE '23:12:12.1267+05:30'", "\"23:12:12.127+05:30\"", CastQuality.LOSSLESS)
```
Comment (lziq) [inline lang/test/org/partiql/lang/eval/CastTestBase.kt:1361]: Why do we want to make this change? L1334-L1346 are not testing symbols as commented in L1333.

Comment (johnedquinn) [reply lang/test/org/partiql/lang/eval/CastTestBase.kt:1361]: It is testing symbols. See the Symbol section of [the Ion Spec](https://amzn.github.io/ion-docs/docs/spec.html#string). TLDR: Symbols can be specified with backticks.

Comment (lziq) [reply lang/test/org/partiql/lang/eval/CastTestBase.kt:1361]: My bad. I thought it was in PartiQL format.

**Review by lziq — APPROVED** [review]

**Pull request merged**